allure-rest-assured/src/test/java/io/qameta/allure/restassured/AllureRestAssuredTest.java

```diff
@@ -23,46 +23,65 @@
+import java.util.List;
 import java.util.Objects;
+import java.util.stream.Collectors;
 import java.util.stream.Stream;
 
 import static io.qameta.allure.test.RunUtils.runWithinTestContext;
 import static org.assertj.core.api.Assertions.assertThat;
+import static org.junit.jupiter.params.provider.Arguments.arguments;
+
+class AttachmentArgumentProvider implements ArgumentsProvider {
+    @Override
+    public Stream<? extends Arguments> provideArguments(ExtensionContext context) {
+
+        return Stream.of(
+                arguments(ImmutableList.of("Request", "HTTP/1.1 200 OK"), new AllureRestAssured()),
+                arguments(ImmutableList.of("Allure Request", "Allure Response"), new AllureRestAssured().setRequestAttachmentName("Allure Request").setResponseAttachmentName("Allure Response")),
+                arguments(ImmutableList.of("Request", "Allure Response"), new AllureRestAssured().setResponseAttachmentName("Allure Response")),
+                arguments(ImmutableList.of("Allure Request", "HTTP/1.1 200 OK"), new AllureRestAssured().setRequestAttachmentName("Allure Request"))
+        );
+    }
+}
 
 /**
  * @author charlie (Dmitry Baev).
  */
 @SuppressWarnings("unchecked")
 class AllureRestAssuredTest {
 
-    static Stream<String> attachmentNameProvider() {
-        return Stream.of("Request", "HTTP/1.1 200 OK");
-    }
-
     @ParameterizedTest
-    @MethodSource(value = "attachmentNameProvider")
-    void shouldCreateAttachment(final String attachmentName) {
+    @ArgumentsSource(AttachmentArgumentProvider.class)
+    void shouldCreateAttachment(final List<String> attachmentNames, AllureRestAssured filter) {
+        RestAssured.replaceFiltersWith(filter);
         final AllureResults results = execute();
 
-        assertThat(results.getTestResults())
-                .flatExtracting(TestResult::getAttachments)
-                .flatExtracting(Attachment::getName)
-                .contains(attachmentName);
+        assertThat(results.getTestResults()
+                .stream()
+                .map(TestResult::getAttachments)
+                .flatMap(Collection::stream)
+                .map(attachment -> attachment.getName()))
+                .isEqualTo(attachmentNames);
     }
 
     @ParameterizedTest
-    @MethodSource(value = "attachmentNameProvider")
-    void shouldCatchAttachmentBody(final String attachmentName) {
+    @ArgumentsSource(AttachmentArgumentProvider.class)
+    void shouldCatchAttachmentBody(final List<String> attachmentNames, AllureRestAssured filter) {
+        RestAssured.replaceFiltersWith(filter);
         final AllureResults results = execute();
-
         final Attachment found = results.getTestResults().stream()
                 .map(TestResult::getAttachments)
                 .flatMap(Collection::stream)
-                .filter(attachment -> Objects.equals(attachmentName, attachment.getName()))
+                .filter(attachment -> attachmentNames.contains(attachment.getName()))
```

Comment:
It doesn't seem like a valid assertion. You're passing a list of names to test, but the actual verification will pass even if there's a single match due to the `findAny` terminal operation call. I believe you just need to extract a collection of attachments from the results and compare 2 lists by content.